### PR TITLE
feat: add a preview parameter to the Notebook launching API [DET-5368]

### PR DESCRIPTION
**Pull request by shiyuann**
## Description

Add a preview parameter to the Notebook launching API. This API returns a full configuration populated with the template and the Notebook configuration.

## Test Plan

- Manually test it.

## Checklist

- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/api_command.go [hunk 1]
```diff
@@ -31,36 +34,122 @@ type protoCommandParams struct {
...
+			return nil, nil, errors.Wrapf(
+				err,
+				"unable to parse the config in the parameters: %s",
+				string(configBytes),
+			)
+		}
+	}
+
+	// mustBeZeroSlot indicates that this type of command may never use more than
+	// zero slots (as of Jan 2021, this is only Tensorboards). This is important
```
Comment (naren-determined) [inline master/internal/api_command.go:71]: Instead of putting a date here, can we put a Determined version?

**Review by naren-determined — APPROVED** [review]
Made one comment that I think should be changed, but approved b/c that is minor and everything else looks good!

File: master/internal/api_command.go (hunk 1 above)
Comment (shiyuann) [reply master/internal/api_command.go:71]: This is a comment that is moved from other places in the codebase. I can't really find the exact version to which it belongs.  Anyways, this should be refactored later. I'd not worry about it

File: master/internal/api_tensorboard.go [hunk 2]
```diff
@@ -69,24 +70,22 @@ func (a *apiServer) LaunchTensorboard(
...
 
 	tensorboardConfig := command.TensorboardRequest{
-		CommandParams: cmdParams,
+		CommandParams: params,
 		ExperimentIDs: experimentIds,
 		TrialIDs:      trialIds,
 	}
-	tensorboardLaunchReq := command.TensorboardRequestWithUser{
-		Tensorboard: tensorboardConfig,
-		User:        user,
```
Comment (hamidzr) [inline master/internal/api_tensorboard.go:88]: question: is the user portion of this type used anywhere else? it seems we can get rid of it.

**Review by hamidzr — APPROVED** [review]

**Review by EmilyBonar — APPROVED** [review]

File: master/internal/api_tensorboard.go (hunk 2 above)
Comment (shiyuann) [reply master/internal/api_tensorboard.go:88]: It is not gotten rid of. It is merged into the `Tensorboard` field. I just reused the same go struct.

Comment (hamidzr) [reply master/internal/api_tensorboard.go:88]: aah I didn't explain it well I meant something like this https://github.com/determined-ai/determined/pull/2377 how does this look?

Comment (shiyuann) [reply master/internal/api_tensorboard.go:88]: It looks good! I approved it

**Pull request merged**